org.eclipse.wildwebdeveloper.tests/src/org/eclipse/wildwebdeveloper/tests/TestESLint.java

```diff
@@ -0,0 +1,110 @@
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.ui.IViewReference;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.IDE;
+import org.eclipse.ui.preferences.ScopedPreferenceStore;
+import org.eclipse.ui.tests.harness.util.DisplayHelper;
+import org.eclipse.ui.texteditor.ITextEditor;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestESLint {
+
+	private IProject project;
+
+	@Before
+	public void setUpProject() throws Exception {
+		ScopedPreferenceStore prefs = new ScopedPreferenceStore(InstanceScope.INSTANCE, "org.eclipse.lsp4e");
+		prefs.putValue("org.eclipse.wildwebdeveloper.jsts.file.logging.enabled", Boolean.toString(true));
+		prefs.putValue("org.eclipse.wildwebdeveloper.css.file.logging.enabled", Boolean.toString(true));
+		prefs.putValue("org.eclipse.wildwebdeveloper.html.file.logging.enabled", Boolean.toString(true));
+		prefs.putValue("org.eclipse.wildwebdeveloper.json.file.logging.enabled", Boolean.toString(true));
+		prefs.putValue("org.eclipse.wildwebdeveloper.xml.file.logging.enabled", Boolean.toString(true));
+		prefs.putValue("org.eclipse.wildwebdeveloper.yaml.file.logging.enabled", Boolean.toString(true));
+		this.project = ResourcesPlugin.getWorkspace().getRoot().getProject(getClass().getName() + System.nanoTime());
+		project.create(null);
+		project.open(null);
+
+		// Setup ESLint configuration and dependencies
+		IFile eslintConfig = project.getFile(".eslintrc");
+		eslintConfig.create(getClass().getResourceAsStream("/testProjects/eslint/.eslintrc"), true, null);
+		IFile tsConfig = project.getFile("tsconfig.json");
+		tsConfig.create(getClass().getResourceAsStream("/testProjects/eslint/tsconfig.json"), true, null);
+		IFile packageJson = project.getFile("package.json");
+		packageJson.create(getClass().getResourceAsStream("/testProjects/eslint/package.json"), true, null);
+		Process dependencyInstaller = new ProcessBuilder(TestAngular.getNpmLocation(), "install")
+				.directory(project.getLocation().toFile()).start();
+		assertEquals("npm install didn't complete properly", 0, dependencyInstaller.waitFor());
+
+		IWorkbenchPage activePage = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+		for (IViewReference ref : activePage.getViewReferences()) {
+			activePage.hideView(ref);
+		}
+	}
+
+	@After
+	public void deleteProjectAndCloseEditors() throws Exception {
+		PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().closeAllEditors(false);
+		this.project.delete(true, null);
+	}
+
+	@Test
+	public void testESLintDiagnostics() throws Exception {
```

Comment:
I tried to run this scenario manually: Import project, run `npm install`, open .ts file, copy it to another file... I could see some plain TypeScript diagnotics because of duplicated variables, but nothing from ESLint.

Reply:
What do you mean by "copy it to another file"? A caveat of ESLint is that the files to be linted must reside within the same directory (or in a child directory) that contains the `.eslintrc`, `tsconfig.json` and the required `node_modules`. If you open a .ts file that isin't in that directory hierarchy, it won't be linted.

Also, did you see if the test was failing?

Reply:
I tried downgrading my `typescript` version to the one that's packaged with Wild Web Developer and it seems to not work with ESLint. Will have to bump up the typescript version.

Reply:
I created a  new file `ESLint2.ts` just beside `ESLint.ts` and coped content and didn't get any linting diagnostic.
Running the test (from IDE as JUnit Plugin Test) fails for me. At the moment, it opens the external text editor, but I'm not sure it's related.

Reply:
So I could get the test opening the right editor, but it's now failing "properly" with _Diagnostic not published_ running both from IDE and with `mvn clean verify`, similarly to how it happens on CI.
Maybe the difference is cause by different node versons. What's yours? mine is `v10.16.3`.